### ISSUE TITLE
Need to merge f549176

### ISSUE THREAD

**Issue by kaz7** [sx-aurora-dev/llvm-project#47]
@simoll -san, can you merge f549176 for me?  It shows following errors.
```
Failed Tests (2):
  LLVM :: Analysis/TypeBasedAliasAnalysis/functionattrs.ll
  LLVM :: Transforms/FunctionAttrs/2008-09-03-ReadOnly.ll
```

I've merged up to 7c02dc2 in #46 which is one predecessor of f549176 belong to upstream/main.

This f549176 patch is reverted multiple times.  I tried to cherry-pick 3c54762 with a hope it solves above problems, but it won't work this time.

**Comment by kaz7** [comment]
In addition, 7c02dc2 itself shows following errors.  These are from mismodifications on VE at upstream.
```
  LLVM :: CodeGen/VE/Scalar/inlineasm-bitconvert.ll
  LLVM :: CodeGen/VE/Scalar/inlineasm-svmlvm2.ll
```
7c02dc2 merged to develop branch shows different following errors.  So, I guess these problems are caued by our mismodifications.
```
  LLVM :: Analysis/TypeBasedAliasAnalysis/functionattrs.ll
  LLVM :: Transforms/FunctionAttrs/2008-09-03-ReadOnly.ll
```

**Comment by kaz7** [comment]
FYI, Our llvm-ve is totally unstable now.  With -DENABLE_EXPERIMENTAL_NEW_PASS_MANAGER=OFF`, we get
```
Failed Tests (2):
  LLVM :: Analysis/TypeBasedAliasAnalysis/functionattrs.ll
  LLVM :: Transforms/FunctionAttrs/2008-09-03-ReadOnly.ll
```
And without it, we get
```
Failed Tests (1):
  LLVM :: CodeGen/Generic/expand-vp.ll
```

**Comment by kaz7** [comment]
This time, I'm going to update `functionattrs.ll` and `2008-09-03-ReadOnly.ll` to make it work with `-DENABLE_EXPERIMENTAL_NEW_PASS_MANAGER=OFF` cmake option.  I hope we can remove it once we merge upstream patches which improve NEW_PM stablness.

**Comment by kaz7** [comment]
Fixed `expand-vp.ll` problem by 8169b6b.
Fixed relative lookup tables problem by c26b4d3.
Now, llvm-ve is working well with `-DENABLE_EXPERIMENTAL_NEW_PASS_MANAGER=ON`.
Need a clean build from scratch, though.
